src/controllers/game_controller.cpp

```diff
@@ -25,23 +26,14 @@ GameController::GameController() {
     createFpsCounter(0);
 }
 
-int GameController::calculateFps(int start_time){
-    int ticks = engine->getTicks();
-    double delta_time_millisecs = ticks - start_time;
-    if (delta_time_millisecs == 0)
-        delta_time_millisecs = 1;
-    if (delta_time_millisecs < fps_frame_time) {
-        double delay = fps_frame_time - delta_time_millisecs;
-        engine->delay(delay);
-        delta_time_millisecs = delta_time_millisecs + delay;
-    }
-
-    delta_time = delta_time_millisecs / 1000;
-
-    // The FPS cannot be calculated correctly because getTicks returns and integer of miliseconds. 
-    // 1000 / 60 = 16.666666, here we lose 0.666 per. But we can only caluclate 1000 / 16 = 62.5. Where 0.5 nd  is lost because of this integer precision issue.
-    double fps = 1000.0 / delta_time_millisecs;
-    return (int)fps;
+int GameController::calculateFps(std::chrono::time_point<std::chrono::high_resolution_clock> start_time){
+    auto end_time = std::chrono::high_resolution_clock::now();
```

Comment:
Is the end_time still 100% accurate when you put it in this function? Shouldn't you create the variable right after drawing the screen inside of the game loop?

Reply:
For performance it will not matter. See this [wiki](https://en.wikipedia.org/wiki/Inline_expansion). For readability I can change the function signature to include both start and end time